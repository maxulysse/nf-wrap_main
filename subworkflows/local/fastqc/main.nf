// FASTQC

include { FASTQC } from '../../../modules/nf-core/fastqc/main.nf'

workflow RUN_FASTQC {
    take:
    input // channel: [mandatory] [ meta, file.fastq.gz ] or [meta, [ file_1.fastq.gz, file_2.fastq.gz ] ]

    main:
    FASTQC(input)

    emit:
    html         = FASTQC.out.html
    zip          = FASTQC.out.zip
    versions     = FASTQC.out.versions
}
