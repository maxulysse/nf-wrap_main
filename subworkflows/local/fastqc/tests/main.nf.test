nextflow_workflow {

    name "Test Workflow RUN_FASTQC"
    script "../main.nf"
    workflow "RUN_FASTQC"
    tag "nf-wrap/fastqc"

    test("FASTQC") {

        when {
            params {
            }
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(file(workflow.out.html[0][1]).name).match("html") },
                { assert snapshot(file(workflow.out.zip[0][1]).name).match("zip") },
                { assert snapshot(workflow.out.versions).match("versions")}
            )
        }
    }
}
