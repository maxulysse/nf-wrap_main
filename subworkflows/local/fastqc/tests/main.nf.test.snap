{
    "zip": {
        "content": [
            "test_fastqc.zip"
        ],
        "timestamp": "2023-11-25T14:09:49.673347"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,0a61fefb0b7384b164c84c6201a3aebc"
            ]
        ],
        "timestamp": "2023-11-25T14:09:49.674722"
    },
    "html": {
        "content": [
            "test_fastqc.html"
        ],
        "timestamp": "2023-11-25T14:09:49.667723"
    }
}